.github/workflows/preview-deployments.yml

```
@@ -16,129 +16,145 @@ env:
 
 jobs:
   macos:
-    runs-on: macos-latest
+    runs-on: macos-15-xlarge
     strategy:
       matrix:
         python-version: ["3.10", "3.11", "3.12", "3.13", "3.14"]
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v3
```

actionlint (1.7.12)

[error] the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)